functions/fenv.main.fish

```diff
@@ -22,33 +22,17 @@
 
 
 function fenv.main
-  set program $argv
-  set divider (fenv.parse.divider)
-  set previous_env (bash -c 'env')
-
-  # Need to ensure that the two calls to env (here and above) have the same
-  # nesting level within bash shells so that the SHLVL variable does not differ.
-  set program_execution (bash -c "$program && echo && echo '$divider' && env" 2>&1)
-  set program_status $status
-
-  if not contains -- "$divider" $program_execution
-    printf '%s\n' $program_execution
-    return $program_status
-  end
-
-  set program_output (fenv.parse.before $program_execution)
-
-  if test $program_status != 0
-    printf "%s\n" $program_output
-    return $program_status
+  bash -c "$argv && printenv -0 >&31" 31>| while read -l -z env_var
```

Comment:
It would be nice if that was supported on macos — although it does on rare cases where `coreutils` is available. So how about not handling newlines in variables?

```suggestion
  bash -c "$argv && printenv >&31" 31>| while read -l env_var
```

Reply:
So `env` on MacOS doesn't support `-0`, right? Maybe I can test `env -0` first. Does it exit with non-zero exit code when called like this?

Reply:
Great! It works! It exits with status `0`. I've updated the suggestion above.

Reply:
> Great! It works! It exits with status `0`. I've updated the suggestion above.

Does it really print all variables on the same line separated by null-byte? Just to make sure. [Here](https://ss64.com/osx/env.html) there is nothing about `-0` flag.

Reply:
```
»»»» man env | head -20
Unknown locale, assuming C
ENV(1)                      General Commands Manual                     ENV(1)

NAME
     env – set environment and execute command, or print environment

SYNOPSIS
     env [-0iv] [-u name] [name=value ...]
     env [-iv] [-P altpath] [-S string] [-u name] [name=value ...]
         utility [argument ...]

DESCRIPTION
     The env utility executes another utility after modifying the environment
     as specified on the command line.  Each name=value option specifies the
     setting of an environment variable, name, with a value of value.  All
     such environment variables are set before the utility is executed.

     The options are as follows:

     -0      End each output line with NUL, not newline.
```

Reply:
Following up with:

```
»»»» bash -c 'env -0 >&31' 31>| while read -z; echo; end | wc -l
      59
```